app/api/v1/auth/index.js

```diff
@@ -82,16 +83,51 @@ router.post('/register', (req, res, next) => {
     return next(new error.BadRequest('Password must be at least 10 characters long'));
   }
 
-  // TODO account activation via email
+
   const newUser = User.sanitize(req.body.user);
   newUser.state = 'ACTIVE';
   User.generateHash(req.body.user.password).then((hash) => {
     newUser.hash = hash;
-    return User.create(newUser);
-  }).then((user) => {
-    log.info('user authentication (registration)', { request_id: req.id, user_uuid: user.uuid });
-    res.json({ error: null, user: user.getPublicProfile() });
-    Activity.accountCreated(user.uuid);
+    // require that we create a user succesfully and send email succesfully
+    db.transaction({
+      isolationLevel: Sequelize.Transaction.ISOLATION_LEVELS.REPEATABLE_READ,
+    }, (transaction) => Promise.all([
+      User.create(newUser).then((user) => {
+        User.generateAccessCode().then((code) => {
+          user.accessCode = code;
+          email.sendEmailVerification(user.email, user.firstName, code).then(() => {
+            // save the access code if email was sent
+            user.save();
```

Comment:
[`Model.prototype.save`](https://sequelize.org/master/class/lib/model.js~Model.html#instance-method-save) returns a Promise so make sure to handle that (or just write to the database once).

Reply:
wrote just once instead, applied same idea to error found in `emailVerification` route